woocommerce/woocommerce-ios
# Blaze: Add picker view for target topics

Closes: #11670 


## Description

This PR adds a new screen for picking target topics for Blaze campaign creation. The implementation is similar to that of target languages and devices, with an exception of the additional header message "Choose relevant topics".

This PR also enhance the experience on `MultiSelectionList`: 
- Hide the `All` row in search mode (when the search query is not empty). This is a workaround to avoid the complicated logic when toggling the All selection in search mode. 
- An additional header "Search result" is also displayed when the search query is not empty to clarify the search mode.

Other changes:
- There's a change in the property name of target topics received from the remote. I updated the Networking model accordingly - thankfully `description` is a protected name in Objective-C so I already used `name` for the attribute in the Core Data entity.
- Updated the state logic on the language picker:
  - check for fetched items before displaying error or loaded state. 
  - removed `debounce` for search query to fix issue with the initial state. we're not making API requests here so it's not necessary.

## Testing instructions

- Log in to a store with at least one existing product and is eligible for Blaze.
- Navigate to the Products tab and select any public product.
- On the product form, select Promote with Blaze.
- Notice that the campaign creation form is presented. Select the Interests field. Confirm the following:
  - Selection works properly.
  - The Interests field on the creation form is updated following the selected items.
  - The Save button is enabled when there is data on the screen, and any item is selected.
  - When entering search queries, the All topics row is not displayed while the "Search results" header is displayed.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/d821cb23-e320-411a-8018-63783a327773




---
